src/layer_builder.rs

```diff
@@ -0,0 +1,278 @@
+use crate::{Chunk, Layer, LayerBundle, LayerSettings, MapTileError, VisibleTile, chunk::ChunkBundle, morton_index, render::TilemapData, round_to_power_of_two, tile::TileBundleTrait};
+use bevy::{
+    prelude::*,
+    render::{
+        mesh::{Indices, VertexAttributeValues},
+        pipeline::PrimitiveTopology,
+    },
+};
+
+/// Useful for creating and modifying a layer in the same system.
+pub struct LayerBuilder<T> {
+    pub settings: LayerSettings,
+    pub(crate) tiles: Vec<(Entity, Option<(T, bool)>)>,
+    pub(crate) layer_entity: Entity,
+}
+
+impl<T> LayerBuilder<T>
+where
+    T: TileBundleTrait,
+{
+    /// Creates the layer builder using the layer settings.
+    pub fn new(commands: &mut Commands, layer_entity: Entity, settings: LayerSettings) -> Self {
```

Comment:
I think that LayerBuilder::new() should take layer_id as a parameter, and we also should probably move layer_id out of LayerSettings since it's not really data that can/should be reusable between layers.

Reply:
I do want people to still be allowed to call build themselves directly on the builder. `map_query.create_layer` is just a nice to have convivence feature.

Reply:
yeah, I didn't notice `build()` yet, then yeah, we should probably add the `layer_id` as a parameter in `LayerBuilder::new()` then probably.

Reply:
We should make sure to add `layer_id` to `new_and_build()` too :)